### PR TITLE
✨ useNativeProcess: a native-listener child-process resource

### PR DESCRIPTION
**Pull request by taras**
# Motivation

Managing a child process previously flowed through `fromReadable` stream subscriptions plus spawned watcher tasks for the `error` and `close` events. That put suspension points between creating the OS process and wiring up its observation and cleanup, and split the lifecycle of a single OS resource across several loosely-coordinated tasks. We want the child process to be a proper resource: acquired synchronously together with its event wiring, torn down through standard cleanup, with platform-specific concerns visibly separated from the common ones.

# Approach

Two commits:

1. **♻️ Feed process stdio from native listeners instead of readable pumps** — `data`/`close` listeners on stdout/stderr push chunks straight into signals, and `error`/`close` listeners on the child resolve the process result, all attached in the same synchronous continuation as the spawn. Replaces the `fromReadable` subscriptions and the `once()` watcher tasks. No behavior change.
2. **✨ Extract `useNativeProcess`** (`src/exec/native.ts`) — a resource that spawns via a caller-provided factory and performs standard cleanup on scope exit: `child.kill()` (SIGTERM) if the process is still alive, then hold the scope open until the OS has reaped the process and closed its pipes. The teardown registers before the child exists, so a halt during acquisition can never find a live process without an armed teardown.

Platform-specific teardown stays in each adapter's own `ensure`, which runs before the resource's backstop: posix keeps the process-group `SIGTERM -pid` and middleware drain; win32 keeps ctrlc, the `Y\n` stdin answer, the drain, and the `killTree` escalation. Whether to fold those into the resource as a termination-request parameter is left as a follow-up design question (it is also where #242's `[force]` symbol would plug in).

Public API and `Stdio` middleware semantics are unchanged, and the existing test suite passes as-is (36/36 × 3 runs; full platform matrix green). One corner case changes deliberately: a posix child that ignores SIGTERM and closes its stdio used to be silently leaked at teardown; the resource now holds the scope until the process actually closes.

## Summary by CodeRabbit

* **Improvements**
  * Improved process execution across POSIX and Windows environments.
  * Standardized handling of process output, errors, completion results, and termination.
  * Added more reliable cleanup when processes are stopped.
  * Improved support for processes with optional input and output streams.
  * Processes without required output streams now report a clear error.

### CODE REVIEW

**Comment by cowboyd** [comment]
It really helps to have the AI generate the PR according to the PR template.

File: process/src/exec/native.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    const spawned = create();
+    child = spawned;
+
+    spawned.once("error", (error) => {
+      result.resolve(Err(error));
+    });
+    spawned.once("close", (code, signal) => {
+      result.resolve(Ok([code ?? undefined, signal ?? undefined]));
+    });
```
Comment (cowboyd) [inline process/src/exec/native.ts]: We should never use `once()` in Effection because un-registering the hook depends on it firing. Instead, event registration should depend solely upon a scope.

```ts
let onError = (error: Error) => result.resolve(Err(error));
spawned.on("error", onError);

try {
  yield* provide({....});
} finally {
  spawned.off("error", onError);
}
```

This is especially important for the I/O pipes which might be shared beyond a single process

**Comment by taras** [comment]
It's really annoying that this can still happen. I'm going to see why this doesn't happen in my other project. Maybe it's because it gets caught in adversarial review.

File: process/src/exec/posix.ts [hunk 2]
```diff
@@ -69,45 +59,37 @@ export function* createPosixProcess(
...
+      let next = yield* subscription.next();
       while (!next.done) {
         yield* Stdio.operations.stderr(next.value);
         stderr.send(next.value);
-        next = yield* io.stderr.next();
+        next = yield* subscription.next();
       }
       stderr.close();
       io.stderrDone.resolve();
     });
```
[the remark this answers is not in the document]
Comment (taras) [reply process/src/exec/posix.ts:83]: Deferring the extraction to #248, which is stacked on this PR and rewrites both of these loops (try/catch/finally around the middleware calls, error routing into processResult). Extracting a helper here would conflict immediately and its signature would change; the dedup belongs where the loops reach their final form.
